src/main/java/sx/blah/discord/handle/obj/IGuild.java

```diff
@@ -543,6 +543,26 @@
 	IEmoji getEmojiByName(String name);
 
 	/**
+	 * Creates a new emoji.
+	 *
+	 * @param name The name, <b>without colons</b> of length 2-32 characters.
+	 * @param image The image of the emoji.
+	 * @param roles The roles for which this emoji will be whitelisted
+	 * @return The new emoji.
+	 */
+	IEmoji createEmoji(String name, Image image, IRole... roles);
+
+	/**
+ 	 * Creates a new emoji.
+ 	 *
+ 	 * @param name The name, <b>without colons</b> of length 2-32 characters.
+ 	 * @param image The image of the emoji.
+ 	 * @return The new emoji.
+ 	 *
+ 	 */
+ 	IEmoji createEmoji(String name, Image image);
```

Comment:
Is this method necessary?

Reply:
While of course this method isn't necessary, 99.9% of bots using D4J will use this method as their bot isn't whitelisted, so specifying roles is useless.